src/tasks/netSdk/NetSdkRunTaskProvider.ts

```diff
@@ -55,4 +51,55 @@ export class NetSdkRunTaskProvider extends DockerTaskProvider {
 
         return Promise.resolve();
     }
+
+    public createNetSdkRunTask(options?: Omit<NetSdkRunTask, "type">): { task: Task, promise: Promise<number> } {
+        let task: Task;
+        const definition = {
+            ...options,
+            type: NetSdkRunTaskType,
+        };
+
+        const promise = new Promise<number>((resolve, reject) => {
+            task = new Task(
+                definition,
+                this.getProjectFolderFromTask(definition),
+                NetSdkDebugTaskName,
+                NetSdkRunTaskType,
+                new CustomExecution(async (resolveDefinition: TaskDefinition) => {
+                    const pseudoTerminal = new DockerPseudoterminal(new NetSdkRunTaskProvider(), task, resolveDefinition);
+
+                    const closeEventRegistration = pseudoTerminal.onDidClose(async (exitCode: number) => {
+                        closeEventRegistration.dispose();
+
+                        if (exitCode === 0) {
+                            resolve(exitCode);
+                        } else {
+                            await NetSdkDebugHelper.clearWorkspaceState();
+                            reject(exitCode);
+                        }
+                    });
+
+                    return pseudoTerminal;
+                }),
+            );
+        });
+
+        return { task, promise };
+    }
+
+    private getProjectFolderFromTask(task: NetSdkRunTask): WorkspaceFolder {
+        if (task?.netCore?.appProject) {
+            const folderUri = Uri.file(path.dirname(task.netCore.appProject));
+            const folder = {
+                uri: folderUri,
+                name: path.basename(folderUri.fsPath),
+                index: 1,
+            };
+            return folder;
+        } else {
+            return workspace.workspaceFolders[0];
+        }
+    }
```

Comment:
We shouldn't need the method as the task should have TaskScope.Workspace set, rather than a specific folder.